cmd/createtree/main.go

```diff
@@ -0,0 +1,194 @@
+//     --pem_key_path=/path/to/pem/file \
+//     --pem_key_password=mypassword
+//
+// The command outputs the tree ID of the created tree to stdout, or an error to
+// stderr in case of failure. The output is minimal to allow for easy usage in
+// automated scripts.
+//
+// Several flags are provided to configure the create tree, most of which try to
+// assume reasonable defaults. Multiple types of private keys may be supported;
+// one has only to set the appropriate --private_key_format value and supply the
+// corresponding flags for the chosen key type.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/sigpb"
+	"google.golang.org/grpc"
+)
+
+var (
+	adminServerAddr = flag.String("admin_server", "", "Address of the gRPC Trillian Admin Server (host:port)")
+
+	treeState          = flag.String("tree_state", trillian.TreeState_ACTIVE.String(), "State of the new tree")
+	treeType           = flag.String("tree_type", trillian.TreeType_LOG.String(), "Type of the new tree")
+	hashStrategy       = flag.String("hash_strategy", trillian.HashStrategy_RFC_6962.String(), "Hash strategy (aka preimage protection) of the new tree")
+	hashAlgorithm      = flag.String("hash_algorithm", sigpb.DigitallySigned_SHA256.String(), "Hash algorithm of the new tree")
+	signatureAlgorithm = flag.String("signature_algorithm", sigpb.DigitallySigned_RSA.String(), "Signature algorithm of the new tree")
+	duplicatePolicy    = flag.String("duplicate_policy", trillian.DuplicatePolicy_DUPLICATES_NOT_ALLOWED.String(), "Duplicate policy of the new tree")
+	displayName        = flag.String("display_name", "", "Display name of the new tree")
+	description        = flag.String("description", "", "Description of the new tree")
+
+	privateKeyFormat = flag.String("private_key_format", "PEMKeyFile", "Type of private key to be used")
+	pemKeyPath       = flag.String("pem_key_path", "", "Path to the private key PEM file")
+	pemKeyPassword   = flag.String("pem_key_password", "", "Password of the private key PEM file")
+)
+
+// createOpts contains all user-supplied options required to run the program.
+// It's meant to facilitate tests and focus flag reads to a single point.
+type createOpts struct {
+	addr                                                                                                      string
+	treeState, treeType, hashStrategy, hashAlgorithm, sigAlgorithm, duplicatePolicy, displayName, description string
+	privateKeyType, pemKeyPath, pemKeyPass                                                                    string
+}
+
+func createTree(ctx context.Context, opts *createOpts) (*trillian.Tree, error) {
+	req, err := newRequest(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := grpc.Dial(opts.addr, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
```

Comment:
You're correct, it returns slightly weird error messages. I'll merge this PR now, as this is blocking other work, but I'll follow up with error msg improvements.